File: ios/WeatherApp/ZipHandler.swift
//
//  ZipHandler.swift
//  WeatherApp
//
//  Created by Matt Chowning on 1/11/18.
//  Copyright © 2018 Matt Chowning. All rights reserved.
//

import Foundation

class ZipHandler {
    
    static func handleInput(input: String) -> String {
        if input.isEmpty {
            return "You entered nothing!"
        } else {
            return "You entered \(input)"
        }
    }
}

File: common/src/main/kotlin/com/mattchowning/ZipHandler.kt
package com.mattchowning


class ZipHandler {

//    fun handleInput(input: String): String =
//            if (input.isEmpty()) "You entered nothing!" else "You entered $input"

    fun handleInput(input: String): String {
        val isValidZipCode = input.toIntOrNull() != null && input.length == 5
        return if (isValidZipCode) {
            "You entered a valid zip code of $input"
        } else {
            "$input is not a valid zip code."
        }
    }
}
